src/librustc_trans/adt.rs

```diff
@@ -556,13 +594,13 @@ fn generic_type_of<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
     }
 }
 
-fn struct_llfields<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>, st: &Struct<'tcx>,
+fn struct_llfields<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>, fields: &Vec<Ty<'tcx>>,
```

Comment:
It falls apart because iterating over a vector directly gives you a `&Ty`, but iterating over a vector and producing a `Ty` i.e. from `monomorphize::field_type` is just `Ty`.  I tried to work it out and couldn't.  If you know what the signature should be, I'm happy to change it.